src/Microsoft.SqlTools.Migration/MigrationService.cs

```diff
@@ -611,15 +611,33 @@ internal async Task HandleGetArmTemplateRequest(
         {
             try
             {
-                Logger.Verbose("request received in toolsservice");
                 ProvisioningScriptServiceProvider provider = new ProvisioningScriptServiceProvider();
                 string searchPattern = $"*{targetType}-Baseline*.json";
                 string skuRecommendationReportFilePath = Directory.GetFiles(SqlAssessmentConfiguration.ReportsAndLogsRootFolderPath, searchPattern).FirstOrDefault();
-                Logger.Verbose($"Logging report file path -- {skuRecommendationReportFilePath}");
-                List<SkuRecommendationResult> recommendations = ExtractSkuRecommendationReportAction.ExtractSkuRecommendationsFromReport(skuRecommendationReportFilePath);
-                Logger.Verbose($"recommendations generated-- {recommendations.Count}");
+
+                // Save the current Console.Out
+                var originalOut = Console.Out;
+                List<SkuRecommendationResult> recommendations;
+                try
+                {
+                    // Redirect Console.Out to a null stream (no output)
+                    using (var nullWriter = new StreamWriter(Stream.Null))
+                    {
+                        Console.SetOut(nullWriter);
+
+                        /* This call was adding some console messages in the response hence the API call was failing in MacOS
+                         * setting Console.Out to a null stream so that this call does not add any random messages to response.*/
+                        recommendations = ExtractSkuRecommendationReportAction.ExtractSkuRecommendationsFromReport(skuRecommendationReportFilePath);
```

Comment:
I have mentioned in PR description we already tried that and still got the same error . 
 Explaining the issue here again -

We found that there was one Console.log line in ExtractSkuRecommendationsFromReport function and we removed that line . PR link - https://msdata.visualstudio.com/Database%20Systems/_git/sql-migrate-asmt/pullrequest/1491680

But even after removing that Console.log line , we encountered same issue on Mac , hence before calling ExtractSkuRecommendationsFromReport method we are setting Console.Out to null so that it does not log any console messages.